src/controllers/tripController.js

```diff
@@ -0,0 +1,211 @@
+        (err, results) => {
+            if (err) {
+                throw err;
+            }
+            const trip_id = results.rows[0].trip_id;
+
+            // Step 3: Insert a record into the user_trip table
+            client.query(
+                `
+                INSERT INTO user_trip (trip_id, user_id)
+                VALUES ($1, $2)`,
+                [trip_id, user_id],
+                (err) => {
+                    if (err) {
+                        throw err;
+                    }
+                });
+            }
+        )
+            
+        const message = "Trip created successfully";
+        res.json({ message });
+    } catch (error) {
+        next(error);
+    }
+}
+
+const UpdateTrip = async (req, res, next) => {
+    const { name, origin, destination, desc, departure_dateTime, arrival_dateTime } = req.body
+    const trip_id = req.params.trip_id;
+    try {
+        client.query(`
+        
+        UPDATE trips SET trip_name = $2, trip_origin = $3, trip_destination = $4,trip_desc = $5, trip_departure_datetime = $6, trip_arrival_datetime = $7
+        WHERE trip_id = $1
+        `,[trip_id, name, origin, destination, desc, departure_dateTime, arrival_dateTime],(err,results)=>{
+            if(err)
+            {
+                throw err;
+            }
+        })
+        res.json({message: "trip updated successfully!"});
+    } catch (error) {
+        next(error);
+    }
+}
+
+const deleteTrip = async (req, res, next) => {
+    const trip_id = req.params.trip_id;
+    try {
+        client.query(`DELETE FROM trips
+        WHERE trip_id = $1`,[trip_id],(err, results) => {
+            if(err){
+                throw err;
+            }
+        })
+        res.json({message: "Trip deleted successfully."})
+    } catch (error) {
+        next(error);
+    }
+}
+
+const queryTripsByCommunityId = async (req, res, next) => {
+    // Collect unique dates, origins, and destinations
+    const tripIds = new Set();
+    const community_id = req.params.community_id;
+    const token = req.header('auth-token');
+    const data = jwt.verify(token, JWT_SECRET);
+    const user_id = data.user.id;
+    
+    try {
+        if(community_id !== undefined)
+        {
+            let sqlQuery = `
+            SELECT DISTINCT CT.trip_id
+            FROM community_trip CT
+            WHERE CT.community_id = ${community_id}
+            `;
+    
+            // Execute the SQL query
+            client.query(sqlQuery, (err, results) => {
+                if (err) {
+                    console.error(err);
+                    return res.status(500).json({ error: 'An error occurred while querying the database' });
+                }
+    
+                results.rows.forEach((row) => {
+                    tripIds.add(row.trip_id);
+                });
+    
+                getTrips(req, res, tripIds);
+            });
+        }
+        else if(community_id === undefined){
+            
```

Comment:
Can you explain this part? its a lot to go through

Reply:
addressed this thing in the recent commit basically it means when the user wants to see without going inside a community

Reply:
Kindly adhere to new changes for this

Reply:
Sorry, I am not getting you.

Reply:
This thing no more exists I have breaked this function in two parts first get all trips second get community only trips.
so the thing on which you have asked question is no more exists see at the new changes in the file

Reply:
okay? to see a community without being in it, the user just needs to make a get request to the community endpoint with the community_id, what had you written was my question.